src/models/logits.cpp

```diff
@@ -20,6 +20,27 @@ Logits::Logits(State& state)
   }
 
   input_sequence_lengths.resize(state_.params_->search.batch_size);
+
+  if (state.model_.p_device_->GetType() == DeviceType::OpenVINO) {
+    const auto& provider_options = state.model_.config_->model.decoder.session_options.provider_options;
+    for (auto& po : provider_options) {
+      if (po.name == "OpenVINO") {
+        const auto& openvino_options = po.options;
+        for (auto& option : openvino_options) {
+          // When 'enable_causallm' option is set, models are patched in a way so that only return the
+          // sliced logits needed for sampling. For example, given 43 prompt tokens, instead of returning
+          // logits of the shape:  [1,43,<vocab_size>]
+          // they will have shape: [1, 1,<vocab_size>].
+          if (option.first == "enable_causallm" && option.second == "True") {
```

Comment:
This check seems to be in multiple places, could we make a utility function for it?

Reply:
Take a look at latest commit, https://github.com/microsoft/onnxruntime-genai/pull/1399/commits/14ac497988815867d0003dd723d79fa1e6c70844